internal/gcsutil/util.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package gcsutil contains utility functions for Google Cloud Storage data access.
+package gcsutil
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+// ContextInStruct contains contexts that can be used in
+// structures when there is no risk of confusion. Using
+// context.Context directly can lead to linting errors.
+type ContextInStruct context.Context
+
+// The documentation for context states:
+//
+//	Contexts should not be stored inside a struct type, but instead passed
+//	to each function that needs it.
+//
+// However, while it is generally important to pass the Context rather than
+// store it in another type, in the case below, this is not needed since there
+// is no risk of confusion. Indeed, the context is only used for the GCS client here.
+// Note that, if the use of context is extended in the future, then it should be
+// passed explicitly to each function that needs it as an argument.
+//
+// Client contains a Google Cloud Storage client and a context.Context.
+type Client struct {
+	StorageClient *storage.Client
+	Context       ContextInStruct
```

Comment:
```suggestion
	context       ContextInStruct
```
Make these fields private (unexposed), since I think no functions outside this package will ever need to access them directly.

Reply:
I agree :)